projects/common/src/time/user-specified-time-range.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Injectable } from '@angular/core';
+import { isNil } from 'lodash-es';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { distinctUntilChanged, map, share } from 'rxjs/operators';
+import { NavigationService } from '../navigation/navigation.service';
+import { PreferenceService, StorageType } from '../preference/preference.service';
+import { TimeRange } from './time-range';
+import { TimeRangeService } from './time-range.service';
+
+@Injectable({ providedIn: 'root' })
+export class UserSpecifiedTimeRangeService {
+  private static readonly STORAGE_TYPE: StorageType = StorageType.Session;
```

Comment:
I think I suggested this since our dev and sales team members frequently switch between tenants and using a session-based storage could avoid any unintended side effects. Thinking again, it doesn't seem a bad idea to use local storage as it would not lead to any failures.

Reply:
Spoke to @anandtiwary about this, I believe it's due to complications around users having multiple account types and switching between them, and possible bugs coming from that.  If that isn't the case however, we can always just flip it to Local if that request comes in.

Reply:
Session vs local is independent of using multiple tenants. Session isn't a login session, it's an application session - the pain of preferences across tenants is the same.